src/sssom/util.py

```diff
@@ -444,33 +444,32 @@ def filter_redundant_rows(df: pd.DataFrame, ignore_predicate: bool = False) -> p
     else:
         key = [SUBJECT_ID, OBJECT_ID, PREDICATE_ID]
     dfmax: pd.DataFrame
-    dfmax = df.groupby(key, as_index=False)[CONFIDENCE].apply(max).drop_duplicates()
-    max_conf: Dict[Tuple[str, ...], float] = {}
-    for _, row in dfmax.iterrows():
+    if not df.empty:
```

Comment:
Should this check be on `df` or `df["confidence"]` and check if `df["confidence"]` is `np.nan` or `''`? 

In the case where this failed in the mondo-ingest build iI believe t was due to the `df["confidence"]` being one of `np.nan` or `''`.

Reply:
it is handled in case of `np.NAN` which was the reason of the error. In the case of `""`, I should not throw any error.

Reply:
Shouldnt we prevent the case that 

https://github.com/mapping-commons/sssom-py/blob/78eacb3a11b08500a1688fdbe4bfe1940fcfef8e/src/sssom/util.py#L441

returns an df.empty in the first place?

Reply:
Not really. The use case itself had all rows for the column `confidence` to be `np.NAN`. `nan_df` becomes the core df of interest an we move forward with it.

Reply:
I am wondering about the case where the original df is not empty, but after cycling through the assign_default_confidence() becomes empty. This would suggest a bug in `assign_default_confidence`. You are saying this cannot happen?

Reply:
It should not become empty. `assign_default_confidence` only adds value to the `confidence` column. There is no scenario where it would convert the entire df to be empty

Reply:
Ahhh ok!